c/tskit/tables.c

```diff
@@ -6742,6 +6747,23 @@ tsk_table_collection_check_mutation_integrity(
                     parent_mut = mutations.parent[parent_mut];
                 }
             }
+            
+            /* Check time ordering, we do this after the time checks above, so that
+               more specific errors trigger first */
+            if (!tsk_is_unknown_time(last_known_time) && 
```

Comment:
It's a bit confusing to have to check ``last_known_time`` is known - could we avoid this by setting ``last_known_time`` to -1 initially or something?

Reply:
We've also already checked for ``!unknown_time`` just above - why not keep this in the same block?

Reply:
Negative times are valid, so using unknown seemed the best way to say "no known time has been seen for this site".

Reply:
Aha, so they are. What about -inf then? This is always less than everything, isn't it, and we check for ``isfinite(time)`` before this.

Reply:
I've used positive infinity (parents before children right?) in https://github.com/tskit-dev/tskit/pull/726/commits/3057d9897f39cd252170a256005f2a41b03cc1e2